paparazzi/src/main/java/app/cash/paparazzi/ImageSize.kt

```diff
@@ -0,0 +1,8 @@
+package app.cash.paparazzi
+
+import app.cash.paparazzi.internal.ImageUtils
+
+sealed class ImageSize {
+  object FullBleed : ImageSize()
+  data class Limit(val height: Int = ImageUtils.THUMBNAIL_SIZE) : ImageSize()
+}
```

Comment:
I think this might be unnecessary a simple `Int` with a `Int.MAX_VALUE` meaning full might be enough? (It's a nice abstraction, but doesn't fit into how the lib is abstracted right now. If it stays...
```suggestion
sealed class ImageSize {
  object Original : ImageSize()
  data class Limit(val height: Int = ImageUtils.THUMBNAIL_SIZE) : ImageSize()
}
```

Reply:
I think using `Int.MAX_VALUE` in that way might be a little confusing. I think consumers would expect the image to scale to the given height. As it would scale down if you pass in a height that is smaller than the original height.

Perhaps a nullable `Int?` and we could name the parameter `maxImageHeight` passing null would mean there is no limit.

What do you think?

https://github.com/cashapp/paparazzi/pull/543/files#r1423206676

Reply:
I don't like `Int?` in general, as no there's a practical limit to max bitmap size. So `maxHeight = 500000` (just a random large number) is the (practically) same as `maxHeight = Int.MAX`.

Reply:
> I think consumers would expect the image to scale to the given height.

That's why I suggested to rename to `maxHeight` because "max" means if it's bigger, resize, otherwise, leave it alone.

Btw, don't let me sway you too much, wait for a Collaborator/Member to make a call here. They might actually prefer the sealed version.

Reply:
Thanks for that. I will hang on making changes for now